cmd/cni-metrics-helper/metrics/pod_watcher.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"context"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PodWatcher interface {
+	GetCNIPods(ctx context.Context) ([]string, error)
+}
+
+type defaultPodWatcher struct {
+	k8sClient client.Client
+	log       logger.Logger
+}
+
+// NewDefaultPodWatcher creates a new podWatcher
+func NewDefaultPodWatcher(k8sClient client.Client, log logger.Logger) *defaultPodWatcher{
+	return &defaultPodWatcher{
+		k8sClient: k8sClient,
+		log: log,
+	}
+}
+
+//Returns aws-node pod info. Below function assumes CNI pods follow aws-node* naming format
+//and so the function has to be updated if the CNI pod name format changes.
+func (d *defaultPodWatcher) GetCNIPods(ctx context.Context) ([]string, error){
+	var CNIPods []string
+	var podList corev1.PodList
+	listOptions := client.ListOptions{
+		Namespace:     metav1.NamespaceSystem,
+	}
+
+	err := d.k8sClient.List(ctx, &podList, &listOptions)
```

Comment:
Please use pagination on all List calls (unless you _know_ the answer is going to be ~1 item)

Reply:
Sure. Since the call is limited to `kube-system` namespace, I used a list here.

Reply:
Ok, I should've clarified in my initial comment but this code is specific to CNI Metrics helper and is not part of `aws-node`. We will only have one CNI Metrics Server pod per cluster + the call being specific to `kube-system` namespace is the reason why a `list` was initially preferred. 

Yeah, I missed that on Limit usage. Considering, the limitation on the cached client end and the call being limited to a single pod + single NS, I'm thinking it should be alright to revert back to no limit usage.

Reply:
> We will only have one CNI Metrics Server pod per cluster

Right, but the response for a 10k cluster is still going to cause ~~100MB to be allocated on apiserver (even for a single List(kube-system) request).  It would be great if we didn't do that.  .. But we might not have a choice until controller-runtime is updated.

Reply:
Scoped it down further using a label selector to filter out aws-node pods and this should help over what we had originally. Verified that it works as expected.

Reply:
Big :+1: to the new label selector approach.  The apiserver thanks you.

Reply:
> If this is a new requirement it should be documented.

Yeah, fair call.  The previous code assumed/required that the pods were all called `aws-node-xxx`, so we're just switching one hard-coded assumption for another.  I agree we should mention that somewhere and/or make the labelselector configurable.

Reply:
@danbeaulieu Default manifest sets this particular label - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/config/v1.7/aws-k8s-cni.yaml#L74.

But agree on documenting this for CNI Metrics helper. If someone wants to use CNI metrics helper than this is required.